library/ssl_tls.c

```diff
@@ -2042,6 +2086,33 @@ static int ssl_tls13_session_load( mbedtls_ssl_session *session,
     memcpy( session->resumption_key, p, session->resumption_key_len );
     p += session->resumption_key_len;
 
+#if defined(MBEDTLS_SSL_SERVER_NAME_INDICATION) && \
+    defined(MBEDTLS_SSL_SESSION_TICKETS) && \
+    defined(MBEDTLS_SSL_CLI_C)
+    if( session->endpoint == MBEDTLS_SSL_IS_CLIENT )
+    {
+        size_t hostname_len;
+        /* load host name */
+        if( end - p < 2 )
+            return( MBEDTLS_ERR_SSL_BAD_INPUT_DATA );
+        hostname_len = MBEDTLS_GET_UINT16_BE( p, 0 );
+        p += 2;
+
+        if( end - p < ( long int )hostname_len )
+            return( MBEDTLS_ERR_SSL_BAD_INPUT_DATA );
+        if( hostname_len > 0 )
+        {
+            session->hostname = mbedtls_calloc( 1, hostname_len + 1 );
```

Comment:
serialized data already include NUL character.
```suggestion
            session->hostname = mbedtls_calloc( 1, hostname_len );
```

Reply:
I am not sure whether there is empty servername only has "\0", if there is not such one, maybe we have no need to wast one extra byte space.
Did someone learn about such cases?

Reply:
I got it in https://github.com/Mbed-TLS/mbedtls/pull/6167/commits/661dd943b6a61ef79d32cff08a96cc9c7e6f850b. the commit is to workaround. 

But the input hostname come from `mbedtls_ssl_set_hostname`, it allows empty string